torch/_inductor/bisect_helper.py

```diff
@@ -0,0 +1,439 @@
+        if not cli_interface:
+            bisection_enabled_orig = cls.bisection_enabled
+            cls.delete_bisect_status()
+            cls.bisection_enabled = True
+
+            class DisableBisect:
+                def __del__(self) -> None:
+                    cls.bisection_enabled = bisection_enabled_orig
+                    cls.delete_bisect_status()
+
+            cleanup = DisableBisect()
+
+        curr_backend = cls.get_backend()
+        curr_subsystem = cls.get_subsystem()
+
+        if not curr_backend:
+            cls.initialize_system()
+            curr_backend = cls.get_backend()
+            curr_subsystem = cls.get_subsystem()
+
+        while True:
+            reset_counters()
+            if curr_subsystem:
+                result = cls.perform_bisection(
+                    curr_backend, curr_subsystem, fn, cli_interface=cli_interface
+                )
+                if result:
+                    curr_subsystem = cls.get_subsystem()
+                    low, _ = cls.get_bisect_range(curr_backend, curr_subsystem)
+                    return BisectionResult(
+                        curr_backend,
+                        curr_subsystem,
+                        low,
+                        call_counter_debug_info.get(low, None),
+                    )
+
+                next_subsystem = cls.advance_subsystem(curr_backend, curr_subsystem)
+                if not next_subsystem:
+                    print(
+                        f"The issue is in the {curr_backend} system, but could not identify subsystem."
+                    )
+                    return BisectionResult(curr_backend)
+
+                curr_subsystem = next_subsystem
+            else:
+                if fn():
+                    next_backend = cls.advance_backend(curr_backend)
+                    if not next_backend:
+                        print("All systems have been checked.")
+                        return None
+
+                    curr_backend = next_backend
+                else:
+                    current_subsystems = BACKENDS[curr_backend]
+                    if current_subsystems:
+                        curr_subsystem = current_subsystems[0]
+                        cls.update_bisect_status(curr_backend, curr_subsystem)
+                        cls.update_run_state(
+                            curr_backend, curr_subsystem, "test_disable"
+                        )
+                        print(
+                            f"The issue is in the {curr_backend} system. Moving to the first subsystem: {curr_subsystem}"
+                        )
+                    else:
+                        print(f"The issue is in the {curr_backend} system.")
+                        return BisectionResult(curr_backend)
+
+            if cli_interface:
+                sys.exit(0)
+
+
+def command_line_usage() -> None:
+    if len(sys.argv) < 2:
+        print("Usage: python bisect_update.py <start|end|good|bad>")
```

Comment:
No way to pass a predicate to automatically tell if it's good/bad?

Reply:
The usage I was envisioning was 

```
python bisect_helper.py start
python benchmarks/dynamo/torchbench.py --performance --no-translation-validation --training --amp --backend inductor --device cuda --only functorch_dp_cifar10
python bisect_helper.py good
...
```

You could hook it up with a script like git bisect but nyi